docs/proposals/20201020-management-cluster-operator.md

```diff
@@ -0,0 +1,844 @@
+Any discrepancies should be rectified in the main Cluster API glossary.
+
+## Summary
+
+The clusterctl CLI currently handles the lifecycle of Cluster API
+providers installed in a management cluster. It provides a great Day 1
+experience in getting CAPI up and running. However, clusterctl’s imperative
+design could make it difficult for cluster admins to stand up and manage CAPI
+management clusters in their own preferred way.
+
+This proposal provides a solution that leverages a declarative API and an
+operator to empower admins to handle the lifecycle of providers within the
+management cluster
+
+## Motivation
+
+In its current form clusterctl is designed to provide a super simple user
+experience for day 1 operations of a Cluster API management cluster with a
+single infrastructure provider.
+
+However such design implicitly is not optimized for supporting declarative
+approaches when operating Cluster API management clusters (e.g. replicating
+clusterctl init is complex).  With the management cluster operator, we aim to
+address these limitations, by introducing an operator that handles the
+lifecycle of providers within the management cluster like clusterctl, just
+based on declarative API.
+
+### Goals
+
+- Define an API that enables declarative management of the lifecycle of
+  Cluster API providers
+- Define and document how providers’ installation and upgrades using the
+  operator should work in air-gapped environments.
+- Define the lifecycle of the operator itself.
+- Identify and document differences between clusterctl CLI and the operator in
+  managing the lifecycle of providers, if any.
+- Define how the clusterctl CLI should be changed in order to interact with
+  the management cluster operator in a transparent and effective way.
+
+### Non-Goals/Future Work
+
+- Deprecate the clusterctl CLI
+- Manage cluster templates using the operator
+- Implement an operator driven version of `clusterctl move`
+- Manage cert-manager using the operator
+- Support multiple installations of the same provider within a management
+  cluster in light of [issue
+  3042](https://github.com/kubernetes-sigs/cluster-api/issues/3042) and [issue
+  3354](https://github.com/kubernetes-sigs/cluster-api/issues/3354)
+
+## Proposal
+
+### User Stories
+
+1. As an admin, I want to use a declarative style API to operate the Cluster
+   API providers in a management cluster.
+1. As an admin, I would like to have an easy and declarative way to change
+   controller settings (e.g. enabling pprof for debugging).
+1. As an admin, I would like to have an easy and declarative way to change the
+   resource requirements such as limits and requests for a provider
+   deployment.
+1. As an admin, I would like to upgrade the Cluster API providers in a
+   management cluster without having to provide the same information I used
+   when I did `clusterctl init`.
+1. As an admin, I would like to have the option to use clusterctl CLI as of
```

Comment:
Is this saying that users can continue to use clusterctl without the operator, or just that they can be unaware it?

Reply:
Moving forward the operator will be required by clusterctl in order to operate. However, the user need not be aware of the existence of the operator. As defined in the Operator Installation, when a user now does a `clusterctl init` on a clean cluster, it will install the operator and then perform similar operations as today.